code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -1,38 +1,10 @@
-/mob/living/carbon/human/update_clothing(slot_flags)
-	if(slot_flags & ITEM_SLOT_BACK)
-		update_worn_back()
-	if(slot_flags & ITEM_SLOT_MASK)
-		update_worn_mask()
-	if(slot_flags & ITEM_SLOT_NECK)
-		update_worn_neck()
-	if(slot_flags & ITEM_SLOT_HANDCUFFED)
-		update_worn_handcuffs()
-	if(slot_flags & ITEM_SLOT_LEGCUFFED)
-		update_worn_legcuffs()
-	if(slot_flags & ITEM_SLOT_BELT)
-		update_worn_belt()
-	if(slot_flags & ITEM_SLOT_ID)
-		update_worn_id()
-	if(slot_flags & ITEM_SLOT_EARS)
-		update_inv_ears()
-	if(slot_flags & ITEM_SLOT_EYES)
-		update_worn_glasses()
-	if(slot_flags & ITEM_SLOT_GLOVES)
-		update_worn_gloves()
-	if(slot_flags & ITEM_SLOT_HEAD)
-		update_worn_head()
-	if(slot_flags & ITEM_SLOT_FEET)
-		update_worn_shoes()
-	if(slot_flags & ITEM_SLOT_OCLOTHING)
-		update_worn_oversuit()
-	if(slot_flags & ITEM_SLOT_ICLOTHING)
-		update_worn_undersuit()
-	if(slot_flags & ITEM_SLOT_SUITSTORE)
-		update_suit_storage()
-	if(slot_flags & (ITEM_SLOT_LPOCKET|ITEM_SLOT_RPOCKET))
-		update_pockets()
-	if(slot_flags & ITEM_SLOT_HANDS)
-		update_held_items()
+/mob/living/carbon/update_obscured_slots(obj/item/worn_item)
+	..()
+	var/obscured_slots = worn_item.flags_inv
+	if(obscured_slots & (HIDEEARS|HIDEEYES|HIDEHAIR|HIDEFACIALHAIR|HIDESNOUT|HIDEMUTWINGS))
+		update_body_parts()
+	if(obscured_slots & HIDEEYES)
+		dna?.species?.handle_body(src)
```

Comment:
handle_body() is called by `update_body()`, just call `update_body()` if any of the flags are there

Reply:
isnt it kinda wasteful to handle body with every single one?

Reply:
update_body_parts() is the most expensive proc in that chain and you want that anyway, its a readability increase for basically no perf loss. this isn't a hot path, dont overthink it

Reply:
alright